src/reactors/PriorityOrderReactor.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.0;
+
+import {BaseReactor} from "./BaseReactor.sol";
+import {Permit2Lib} from "../lib/Permit2Lib.sol";
+import {PriorityOrderLib, PriorityOrder, PriorityInput, PriorityOutput} from "../lib/PriorityOrderLib.sol";
+import {PriorityFeeLib} from "../lib/PriorityFeeLib.sol";
+import {SignedOrder, ResolvedOrder} from "../base/ReactorStructs.sol";
+import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
+
+/// @notice Reactor for simple limit orders
+contract PriorityOrderReactor is BaseReactor {
+    using Permit2Lib for ResolvedOrder;
+    using PriorityOrderLib for PriorityOrder;
+    using PriorityFeeLib for PriorityInput;
+    using PriorityFeeLib for PriorityOutput[];
+
+    error InvalidDeadline();
+    error OrderNotFillable();
+    error InputOutputScaling();
+
+    constructor(IPermit2 _permit2, address _protocolFeeOwner) BaseReactor(_permit2, _protocolFeeOwner) {}
+
+    /// @inheritdoc BaseReactor
+    function _resolve(SignedOrder calldata signedOrder)
+        internal
+        view
+        override
+        returns (ResolvedOrder memory resolvedOrder)
+    {
+        PriorityOrder memory priorityOrder = abi.decode(signedOrder.order, (PriorityOrder));
+        _validateOrder(priorityOrder);
+
+        uint256 priorityFee = tx.gasprice - block.basefee;
```

Comment:
one thing to make sure we consider is how different L2s handle these fields - I assume the default handling is the same as mainnet eip-1559 but potentially some handle differently due to separation of L1 data fee, slightly different gas rules etc.
Worth adding a comment to the contract I think clarifying that this only works on chains that have xyz properties

Reply:
oooo good callout, so it's possible that this calculated value could include other additional fees / not represent the actual priority fee correctly?